tests/extension/oneapi_kernel_compiler_spirv/kernel_compiler_spirv.cpp

```diff
@@ -0,0 +1,235 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2023 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#include "../../common/common.h"
+#include <exception.hpp>
+
+namespace kernel_compiler_spirv::tests {
+
+#ifdef SYCL_EXT_ONEAPI_AUTO_LOCAL_RANGE
+
+sycl::kernel_bundle<sycl::bundle_state::executable> loadKernelsFromFile(
+    sycl::queue& q, std::string file_name) {
+  namespace syclex = sycl::ext::oneapi::experimental;
+
+  // Read the SPIR-V module from disk.
+  std::ifstream spv_stream(file_name, std::ios::binary);
```

Comment:
I am a bit uncomfortable with the case where the file is not found, which might not be obvious to diagnose.
I do not know whether https://en.cppreference.com/w/cpp/filesystem/exists and/or https://en.cppreference.com/w/cpp/filesystem/file_size would make the code better.